doc/reset_factory/ResetFactoryHLD.md

```diff
@@ -0,0 +1,210 @@
+In addition, it will clear logs, tech-support and reboot-cause files, users history files and home directories.<br/>
+
+Steps:
+
+1. Stops sonic.target: 
+
+systemctl stop sonic.target
+
+2. If not "keep-all-config":
+
+config-setup factory <None/keep-basic/only-config>
+
+3. If not ("only-config" or “keep-basic”):
+
+* Delete bash , vim and python history files under "/home".
+* Delete any non-dotfiles in the home directories.
+
+4. If not "only-config":
+* Delete tech-support files under "/var/dump/"
+* Delete all files under "/var/log"
+* Delete all files and symlinks under "/host/reboot-cause/"
+
+5. Print to /var/log/systemlog:
+New log file that won't be cleared in reset-factory.
+It will allow user to track operations such as "reset-factory".
+
+6. Reboot
+
+
+## 3 <a name='Functionality'></a>Functionality
+###  3.1 <a name='Reset-Factory'></a>Reset-Factory
+
+##### Default
+It will generate the default config_db.json file by running command:<br/>
+config-setup factory<br/>
+It will clear system logs and files.
+
+##### Keep-basic
+
+It will generate the default config_db.json and merge it with basic configurations from the current “config_db.json”.<br/>
+Command: config-setup factory keep-basic<br/>
+It will clear system logs and files.<br/>
+
+##### Keep-all-config
+It will keep the same configuration but will clear system logs and files.
+
+##### Only-config
+It will generate the default config_db.json file by running command:<br/>
+config-setup factory only-config<br/>
+It won't clear system logs and files.
+
+###  3.2 <a name='Config-Setup-Factory'></a>Config-setup Factory
+
+config-setup script provides the functionality of creating factory default configuration on demand.<br/>
+Currently it generates new config_db.json file using:<br/>
+```
+sonic-cfggen -H -k ${HW_KEY} --preset ${DEFAULT_PRESET}
+```
+
+Also, it allows the user to extend/replace the functionality by adding hooks to the config-setup directory.
+
+We will extend config-setup factory to support new factory type parameter (keep-basic).
+This parameter will be passed to the hook scripts so user can extend/replace those flows also.
+
+We will add a new conf file for config-setup: /etc/config-setup/config-setup.conf.
+The file will be imported in the script and it will store variables and configurations of the script.
+
+##### Keep-basic
+
+It will generate the default config_db.json merged with basic configurations from the current “config_db.json”<br/>
+The list of tables we will be stored in config-setup.conf:<br/>
+```
+KEEP_BASIC_TABLES=MGMT_PORT,MGMT_INTERFACE,PASSW_HARDENING
+```
+
+We will extend this list with the following tables after the merge of features:
+```
+SSH_SERVER,USER_TABLE,ROLE_TABLE
+```
+
+
+## 4 <a name='CLI'></a>CLI
+
+#### reset-factory
+```
+==============================================================================
+root@host:~$ sudo reset-factory --help
+ Usage:  reset-factory < keep-all-config | only-config | keep-basic >
```

Comment:
Add 'none' or 'default' option as well.

Reply:
I think it would be confusing to print something like:
`Usage:  reset-factory < none | keep-all-config | only-config | keep-basic >`
I think it will be better to explain about the default flow before the parameters (line 185)